### DISCUSSION TITLE
Update .net version 6.0 to 8.0

### DISCUSSION THREAD

**Discussion by RutulPatel8** [dotnet/aspnetcore#54567] [General] [answered]
when I update .net core version from 6.0 to 8.0. I might affect the api response. due to that It is breaking whole my application.
Anyone can help to sort these.

Here is the 6.0 response :
{[
    {
      "date": "2024-03-15T12:47:35.4791006+05:30",
      "temperatureC": -14,
      "temperatureF": 7,
      "summary": "Scorching"
    },
    {
      "date": "2024-03-16T12:47:35.4791059+05:30",
      "temperatureC": 2,
      "temperatureF": 35,
      "summary": "Sweltering"
    },
    {
      "date": "2024-03-17T12:47:35.479106+05:30",
      "temperatureC": -2,
      "temperatureF": 29,
      "summary": "Bracing"
    },
    {
      "date": "2024-03-18T12:47:35.4791061+05:30",
      "temperatureC": 29,
      "temperatureF": 84,
      "summary": "Cool"
    },
    {
      "date": "2024-03-19T12:47:35.4791062+05:30",
      "temperatureC": 23,
      "temperatureF": 73,
      "summary": "Balmy"
    }
  ]}
  
  Here is 8.0 Response :
  {
  "version": "1.0",
  "message": "GET Request successful.",
  "result": [
    {
      "date": "2024-03-15T12:47:35.4791006+05:30",
      "temperatureC": -14,
      "temperatureF": 7,
      "summary": "Scorching"
    },
    {
      "date": "2024-03-16T12:47:35.4791059+05:30",
      "temperatureC": 2,
      "temperatureF": 35,
      "summary": "Sweltering"
    },
    {
      "date": "2024-03-17T12:47:35.479106+05:30",
      "temperatureC": -2,
      "temperatureF": 29,
      "summary": "Bracing"
    },
    {
      "date": "2024-03-18T12:47:35.4791061+05:30",
      "temperatureC": 29,
      "temperatureF": 84,
      "summary": "Cool"
    },
    {
      "date": "2024-03-19T12:47:35.4791062+05:30",
      "temperatureC": 23,
      "temperatureF": 73,
      "summary": "Balmy"
    }
  ]
}

**Comment by markmouton22** [comment]
Hi :) @RutulPatel8 

 I used this to upgrade some projects from dotnet 6 to 8 last week:  https://learn.microsoft.com/en-us/dotnet/core/porting/upgrade-assistant-overview?WT.mc_id=dotnet-35129-website
 
 Worked great for me :)
 

-  Maybe try reverting before the upgrade and then use the upgrade assistant.

 
 I see that the structure of the API response has changed when you upgraded from dotnet core 6 to 8. In the dotnet 6 version, the response was a simple array of objects, but in the dotnet 8 version, the response is an object with additional fields (version, message, and result), where result is the array of objects.

**Reply by RutulPatel8** [reply, marked as answer]
Yes, I see but it is due to the response wrapper but it works well for me.
